### PR TITLE
 #137 refactored RqMethod into Interface

### PR DESCRIPTION
**Pull request by MohandAndel**
created RqMethod.Base

### CODE REVIEW

**Comment by MohandAndel** [comment]
@davvd  please someone review this

**Comment by davvd** [comment]
@mohandandel Thanks, I will ask someone to do a code review here, thanks for your pull request

**Comment by davvd** [comment]
@krzyk this pull request is for you, please review

**Comment by krzyk** [comment]
@MohandAndel Please add a reference to the original issue in the description. Also add  a short sentence or two description of what you did in this PR (something more than *implemented method abc()*).

File: src/main/java/org/takes/rq/RqMethod.java [hunk 1]
```diff
@@ -29,69 +29,78 @@
...
-        final String[] parts = line.split(" ", 2);
-        return parts[0].toUpperCase(Locale.ENGLISH);
+    String method() throws IOException;
+
+    @EqualsAndHashCode(callSuper = true)
+    final class Base extends RqWrap implements RqMethod {
+
+        /**
+         * Ctor.
+         *
```
Comment (krzyk) [inline src/main/java/org/takes/rq/RqMethod.java]: @MohandAndel no need for additional new line between method description and params

File: src/main/java/org/takes/rq/RqMethod.java [hunk 2]
```diff
@@ -29,69 +29,78 @@
...
-     * @param req Original request
-     */
-    public RqMethod(final Request req) {
-        super(req);
-    }
+    String PATCH = "PATCH";
 
     /**
      * Get method.
+     *
```
Comment (krzyk) [inline src/main/java/org/takes/rq/RqMethod.java]: @MohandAndel please remove this additional new line

**Comment by krzyk** [comment]
@MohandAndel Thanks. Please see my styling comments above.

**Comment by krzyk** [comment]
@MohandAndel you have to merge master

**Comment by krzyk** [comment]
@MohandAndel please close this PR if you are not longer working on it

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by elenavolokhova** [comment]
@krzyk 
Thank you for the work on quality of this ticket! 
As far as it wasn't merged there is no reason to fix it's description now. In the future try to be more persistent and don't complete the review until such issues are fixed.
One more comment. There were only two issues found in the code. Try to find at least three bugs next time.
We are on the right way :)

**Comment by krzyk** [comment]
@elenavolokhova thank you,  I'll do better next time

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@elenavolokhova thanks for the review, we'll try better next time

**Comment by davvd** [comment]
@krzyk just added 10 mins to @elenavolokhova (for QA), payment ID is `56378499`

Thank you, I have added **20 mins** to your account in payment/transaction "`56378516`", time consumed: 263 hours and 20 mins

you're getting extra minutes here (c=5)

+20 added to your rating, current score is: [+5074](http://www.netbout.com/b/35035?open=rating)

**Pull request closed without merging**